src/tests/roc_core/test_pool.cpp

```diff
@@ -479,5 +479,58 @@ TEST(pool, embedded_capacity_reuse) {
     LONGS_EQUAL(0, arena.num_allocations());
 }
 
+TEST(pool, boundary_guard_around_object) {
+    TestArena arena;
+    Pool<TestObject, 1> pool("test", arena);
+    void* pointer = NULL;
+
+    pointer = pool.allocate();
+    CHECK(pointer);
+
+    char* data = (char*)pointer;
+    char* before_data = data - 1;
+    char* after_data = data + AlignOps::align_max(sizeof(TestObject));
+    // See PoisonOps::Pattern_BoundaryGuard.
+    char Pattern_BoundaryGuard = 0x7b;
+    CHECK(*before_data == Pattern_BoundaryGuard);
+    CHECK(*after_data == Pattern_BoundaryGuard);
+
+    pool.deallocate(pointer);
+}
+
+IGNORE_TEST(pool, panics_on_boundary_guard_before_embedded_object_violation) {
+    TestArena arena;
+    Pool<TestObject, 1> pool("test", arena);
+    void* pointer = NULL;
+
+    pointer = pool.allocate();
+    CHECK(pointer);
+
+    char* data = (char*)pointer;
+    data--;
+    *data = 0x00;
+
+    pool.deallocate(pointer);
+}
+
+IGNORE_TEST(pool, panics_on_boundary_guard_after_non_embedded_object_violation) {
```

Comment:
The name is a bit too long and hard to read, maybe use something shorter like `guard_before_object` and `guard_after_object`?